### PR TITLE
Modifies GaussianProcessRegressor to use cdist custom operator if option optim='cdist'

### PR DESCRIPTION
**Pull request by xadupre**

### CODE REVIEW

File: skl2onnx/operator_converters/_gp_kernels.py [hunk 1]
```diff
@@ -74,21 +87,34 @@ def py_make_float_array(cst, dtype, as_tensor=False):
 
 
 def _convert_exp_sine_squared(X, Y, length_scale=1.2, periodicity=1.1,
-                              pi=3.141592653589793, dtype=None, **kwargs):
-    dists = onnx_cdist(X, Y, metric="euclidean", dtype=dtype)
+                              pi=3.141592653589793, dtype=None, optim=None,
```
Comment (prabhat00155) [inline skl2onnx/operator_converters/_gp_kernels.py]: Should we use math.pi instead?

File: tests/test_sklearn_gaussian_process.py [hunk 2]
```diff
@@ -81,6 +82,7 @@
 
 
 threshold = "0.4.0"
+threshold2 = "0.5.0"
```
Comment (prabhat00155) [inline tests/test_sklearn_gaussian_process.py]: Constants should be in capital letters.

File: skl2onnx/algebra/complex_functions.py [hunk 3]
```diff
@@ -9,85 +9,144 @@
...
     elif metric == 'euclidean':
-        res = _onnx_cdist_sqeuclidean(X, Y, dtype=dtype)
-        return OnnxSqrt(res, **kwargs)
+        res = _onnx_cdist_sqeuclidean(X, Y, dtype=dtype, op_version=op_version)
+        return OnnxSqrt(res, op_version=op_version, **kwargs)
+    elif metric == 'minkowski':
+        p = kwargs.pop('p')
+        res = _onnx_cdist_minkowski(
+            X, Y, dtype=dtype, op_version=op_version, p=p)
+        return OnnxPow(res, np.array([1. / p], dtype=dtype),
```
[the remark this answers is not in the document]
Comment (xadupre) [reply skl2onnx/algebra/complex_functions.py]: No unless the model was not trained.

File: skl2onnx/operator_converters/_gp_kernels.py (hunk 1 above)
Comment (xadupre) [reply skl2onnx/operator_converters/_gp_kernels.py]: It should be better.

File: tests/test_sklearn_gaussian_process.py (hunk 2 above)
Comment (xadupre) [reply tests/test_sklearn_gaussian_process.py]: done

File: tests/test_algebra_onnx_operators_scan.py [hunk 4]
```diff
@@ -29,6 +30,7 @@
 
 
 threshold = "0.4.0"
+threshold2 = "0.5.0"
```
Comment (prabhat00155) [inline tests/test_algebra_onnx_operators_scan.py]: Again this is constant, should be in capitals.

**Pull request merged**